services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocol.java

```diff
@@ -384,6 +384,9 @@ public Response frontchannelLogout(UserSessionModel userSession, AuthenticatedCl
     @Override
     public Response finishBrowserLogout(UserSessionModel userSession, AuthenticationSessionModel logoutSession) {
         event.event(EventType.LOGOUT);
+        if (event.getEvent().getClientId() == null) {
```

Comment:
Not sure if we need the check but just set the client from the authentication session?

Reply:
This is minor, but yes, I think `if` can be removed.

Reply:
I didn't want to overwrite anything that came from somewhere else. But if you think this is not needed, I will remove it.

Reply:
@Delawen The way you did is safer but I can't see why a client other than the one associated with the authentication session should be set there. Not hard on it as we can figure it out later.

Reply:
Thanks @Delawen !